kong/conf_loader.lua

```diff
@@ -167,29 +209,150 @@ local CONF_INFERENCES = {
   db_cache_ttl = {  typ = "number"  },
   db_resurrect_ttl = {  typ = "number"  },
   db_cache_warmup_entities = { typ = "array" },
-  nginx_user = { typ = "string" },
-  nginx_worker_processes = { typ = "string" },
+  nginx_daemon = { -- TODO: remove since deprecated in 2.0
+    typ = "ngx_boolean",
+    deprecated = {
+      replacement = "nginx_main_daemon",
+      alias = function(conf)
+        if conf.nginx_main_daemon == nil then
+          if type(conf.nginx_daemon) == "boolean" then
+            if conf.nginx_daemon then
+              conf.nginx_main_daemon = "on"
+            else
+              conf.nginx_main_daemon = "off"
+            end
+
+          else
+            conf.nginx_main_daemon = tostring(conf.nginx_daemon)
+          end
+        end
+      end,
+    }
+  },
+  nginx_user = { -- TODO: remove since deprecated in 2.0
+    typ = "string",
+    deprecated = {
+      replacement = "nginx_main_user",
+      alias = function(conf)
+        if conf.nginx_main_user == nil then
```

Comment:
If `if conf.nginx_main_user ~= nil`, should we error out and let the user know they can not configure both at the same time. Right now this will silently make `nginx_user` a no-op.

Reply:
It will not make it `silently` `WARN` will be logged to both `CLI` and NGINX logs.